api/handlers.go

```diff
@@ -330,6 +330,39 @@ func getVolume(c *context, w http.ResponseWriter, r *http.Request) {
 	httpError(w, fmt.Sprintf("No such volume: %s", name), http.StatusNotFound)
 }
 
+func GetUsedVolumes(containers []*cluster.Container) map[string]bool {
+	usedVolumes := make(map[string]bool)
+	for _, container := range containers {
+		for _, mount := range container.Mounts {
+			if mount.Type == "volume" {
+				usedVolumes[container.Engine.Name+"/"+mount.Name] = true
+			}
+		}
+	}
+	return usedVolumes
+}
+
+func GetFilteredVolumes(containers []*cluster.Container, volumes []*apitypes.Volume, dangling bool) []*apitypes.Volume {
```

Comment:
As above, make this function private.

Reply:
right, the unit test is in the same package, no need for public.
I will fix that